iceoryx_hoofs/test/moduletests/test_vocabulary_semantic_string.cpp

```diff
@@ -67,6 +74,156 @@ template <>
 const std::vector<std::string> TestValues<UserName>::INVALID_CONTENT_ADD_BEGIN{"-bla", "81923"};
 template <>
 const std::vector<std::string> TestValues<UserName>::INVALID_CONTENT_ADD_END{};
+///////////////////
+// END: UserName
+///////////////////
+
+///////////////////
+// START: GroupName
+///////////////////
+template <>
+const uint64_t TestValues<GroupName>::CAPACITY = platform::MAX_GROUP_NAME_LENGTH;
+template <>
+const std::vector<std::string> TestValues<GroupName>::VALID_VALUES{{"a-group"}, {"group2"}};
+template <>
+const std::vector<std::string> TestValues<GroupName>::INVALID_CHARACTER_VALUES{
+    {"se-!ur"}, {"*kad"}, {"_fus"}, {"a/sd"}, {";'1'fu"}, {"ah/"}, {"fuu/bla"}};
+template <>
+const std::vector<std::string> TestValues<GroupName>::INVALID_CONTENT_VALUES{{""}, {"-no-dash"}, {"5no-number"}};
+template <>
+const std::vector<std::string> TestValues<GroupName>::TOO_LONG_CONTENT_VALUES{{"i-am-waaaaay-toooooooo-loooooooong"}};
+template <>
+const std::string TestValues<GroupName>::GREATER_VALID_VALUE{"zebra-zusel"};
+template <>
+const std::string TestValues<GroupName>::SMALLER_VALID_VALUE{"alfons-alf"};
+template <>
+const std::string TestValues<GroupName>::MAX_CAPACITY_VALUE{"all-glory-to-the-incredible-and-legendary-hypno-toad"};
+template <>
+const std::vector<std::string> TestValues<GroupName>::INVALID_CONTENT_ADD_BEGIN{"-fuu", "8number"};
+template <>
+const std::vector<std::string> TestValues<GroupName>::INVALID_CONTENT_ADD_END{};
+///////////////////
+// END: GroupName
+///////////////////
+
+///////////////////
+// START: FileName
+///////////////////
+template <>
+const uint64_t TestValues<FileName>::CAPACITY = platform::IOX_MAX_FILENAME_LENGTH;
+template <>
+const std::vector<std::string> TestValues<FileName>::VALID_VALUES{
+    {"file"}, {"another_file.bla"}, {"123.456"}, {".hidden_me"}};
+template <>
+const std::vector<std::string> TestValues<FileName>::INVALID_CHARACTER_VALUES{
+    {"some-!user"}, {"*kasjd"}, {"$_fuuuas"}, {"asd/asd"}, {";'1'fuuuu"}, {"argh/"}, {"fuu/arg/bla"}};
+template <>
+const std::vector<std::string> TestValues<FileName>::INVALID_CONTENT_VALUES{{""}, {"."}, {".."}};
+template <>
+const std::vector<std::string> TestValues<FileName>::TOO_LONG_CONTENT_VALUES{
+    std::string(platform::IOX_MAX_FILENAME_LENGTH + 2, 'a')};
+template <>
+const std::string TestValues<FileName>::GREATER_VALID_VALUE{"9-i-am-a-file"};
+template <>
+const std::string TestValues<FileName>::SMALLER_VALID_VALUE{"0.me.too.be.file"};
+template <>
+const std::string TestValues<FileName>::MAX_CAPACITY_VALUE{std::string(platform::IOX_MAX_FILENAME_LENGTH, 'b')};
+template <>
+const std::vector<std::string> TestValues<FileName>::INVALID_CONTENT_ADD_BEGIN{};
+template <>
+const std::vector<std::string> TestValues<FileName>::INVALID_CONTENT_ADD_END{};
```

Comment:
Just wondering. Shouldn't these contain some of the values from `INVALID_CHARACTER_VALUES`/`INVALID_CONTENT_VALUES`?

Reply:
I renamed the variable to make it clearer.